libkbfs/block_retrieval_queue.go

```diff
@@ -154,17 +166,32 @@ func (brq *blockRetrievalQueue) popIfNotEmpty() *blockRetrieval {
 }
 
 // notifyWorker notifies workers that there is a new request for processing.
-func (brq *blockRetrievalQueue) notifyWorker() {
+func (brq *blockRetrievalQueue) notifyWorker(priority int) {
+	workerCh := brq.workerCh
+	if priority < defaultOnDemandRequestPriority {
+		workerCh = brq.prefetchWorkerCh
+	}
 	select {
 	case <-brq.doneCh:
 		retrieval := brq.popIfNotEmpty()
 		if retrieval != nil {
 			brq.FinalizeRequest(retrieval, nil, io.EOF)
 		}
-	// Get the next queued worker
-	case ch := <-brq.workerQueue:
-		retrieval := brq.popIfNotEmpty()
-		ch <- retrieval
+	// Notify the next queued worker.
+	case workerCh <- struct{}{}:
+	default:
+		// Only launch a goroutine if we would otherwise block.
+		go func() {
+			select {
+			case <-brq.doneCh:
+				retrieval := brq.popIfNotEmpty()
```

Comment:
Hah I was afraid someone was going to suggest that.

It's 4 lines, but I can wrap it. I can't wrap the whole `select`, because the first one has a `default` and the second one does not.